aggregator_test.go

```diff
@@ -124,23 +128,38 @@ func TestSigmoidNorm_Normalize(t *testing.T) {
 
 func TestReverseMinNorm_Normalize(t *testing.T) {
 	t.Run("reverseMin norm should not panic", func(t *testing.T) {
-		norm := &reverseMinNorm{min: 0}
+		norm := NewReverseMinNorm(0)
 		require.NotPanics(t, func() { norm.Normalize(0) })
 
-		norm = &reverseMinNorm{min: 1}
+		norm = NewReverseMinNorm(1)
 		require.NotPanics(t, func() { norm.Normalize(0) })
 	})
 
 	t.Run("reverseMin norm should equal 1 at min value", func(t *testing.T) {
-		norm := &reverseMinNorm{min: 10}
+		norm := NewReverseMinNorm(10)
+		require.InEpsilon(t, 1.0, norm.Normalize(10), eps)
+	})
+}
+
+func TestMaxNorm_Normalize(t *testing.T) {
+	t.Run("max norm should not panic", func(t *testing.T) {
+		norm := NewMaxNorm(0)
+		require.NotPanics(t, func() { norm.Normalize(1) })
+
+		norm = NewMaxNorm(1)
+		require.NotPanics(t, func() { norm.Normalize(0) })
+	})
+
+	t.Run("max norm should equal 1 at max value", func(t *testing.T) {
+		norm := NewMaxNorm(10)
 		require.InEpsilon(t, 1.0, norm.Normalize(10), eps)
 	})
 }
 
 func TestBucket_TraverseTree(t *testing.T) {
 	var (
-		meanAF = AggregatorFactory{New: func() Aggregator { return new(meanAgg) }}
```

Comment:
There is no need to have an additional call here. It is ok not to use public constructors in internal implementation.

Reply:
In the test I wanted to cover publicly available functions as more as I can.

Reply:
Could you then move this changes into a separate commit?

Reply:
Otherwise, LGTM.